### ISSUE TITLE
oceMagic() errors with multibyte strings

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1097]
Example:
```R
library(oce)
f<-"/Users/kelley/Dropbox/oce-working-notes/tests/adp-rdi/vmdas_workhorse_300kHz_01/ADCP061_000000.NMS"
oceMagic(f)
```
yields
```R
 Error in substr(line, 1, 3) : 
  invalid multibyte string at '<a5>b...
```
PS. amusingly, in R.app the bell beeps ... but in Rstudio, the console window becomes unscrollable. Good one, Rstudio.

**Comment by dankelley** [comment]
The problem could be solved by messing with how the line is read, but I don't want to do that. I prefer to switch to using `grep`, e.g.
```R
    if (1 == length(grep("^CTD", line, useBytes=TRUE))) { #substr(line, 1, 3) == "CTD") {
```
where the comment is the old way. Therefore I am going to do this for all the `substr` calls in `R/oce.R` and will do a bit of testing before I push to GH.

**Comment by dankelley** [comment]
Fixed in `develop` commit df5d0edc66f3830c03d12b6e7c71e845a46f236a
